velox/functions/sparksql/specialforms/DecimalRound.cpp

```diff
@@ -0,0 +1,200 @@
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& resultType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const {
+    VELOX_USER_CHECK(
+        args[0]->isConstantEncoding() || args[0]->isFlatEncoding(),
+        "Single-arg deterministic functions receive their only argument as flat or constant vector.");
+    auto rawResults = prepareResults(rows, resultType, context, result);
+    if (args[0]->isConstantEncoding()) {
+      // Fast path for constant vectors.
+      auto constant = args[0]->asUnchecked<ConstantVector<A>>()->valueAt(0);
+      context.applyToSelectedNoThrow(
+          rows, [&](auto row) { rawResults[row] = applyRound(constant); });
+    } else {
+      // Fast path for flat.
+      auto rawA = args[0]->asUnchecked<FlatVector<A>>()->mutableRawValues();
+      context.applyToSelectedNoThrow(
+          rows, [&](auto row) { rawResults[row] = applyRound(rawA[row]); });
+    }
+  }
+
+  inline R applyRound(const A input) const {
+    if (scale_ >= 0) {
+      auto rescaledValue = DecimalUtil::rescaleWithRoundUp<A, R>(
+          input, inputPrecision_, inputScale_, resultPrecision_, resultScale_);
+      VELOX_DCHECK(rescaledValue.has_value());
+      return rescaledValue.value();
+    } else {
+      auto rescaleFactor = DecimalUtil::kPowersOfTen[inputScale_ - scale_];
+      R rescaledValue;
+      DecimalUtil::divideWithRoundUp<R, A, int128_t>(
+          rescaledValue, input, rescaleFactor, false, 0, 0);
+      rescaledValue *= DecimalUtil::kPowersOfTen[-scale_];
+      return rescaledValue;
+    }
+  }
+
+  const int32_t scale_;
+  const uint8_t inputPrecision_;
+  const uint8_t inputScale_;
+  const uint8_t resultPrecision_;
+  const uint8_t resultScale_;
+};
+
+std::shared_ptr<exec::VectorFunction> createDecimalRound(
+    int32_t scale,
+    const TypePtr& inputType,
+    const TypePtr& resultType) {
+  auto [inputPrecision, inputScale] = getDecimalPrecisionScale(*inputType);
+  auto [resultPrecision, resultScale] = getDecimalPrecisionScale(*resultType);
+  if (inputType->isShortDecimal()) {
+    if (resultType->isShortDecimal()) {
+      return std::make_shared<DecimalRoundFunction<int64_t, int64_t>>(
+          scale, inputPrecision, inputScale, resultPrecision, resultScale);
+    } else {
+      return std::make_shared<DecimalRoundFunction<int128_t, int64_t>>(
+          scale, inputPrecision, inputScale, resultPrecision, resultScale);
+    }
+  } else {
+    if (resultType->isShortDecimal()) {
+      return std::make_shared<DecimalRoundFunction<int64_t, int128_t>>(
+          scale, inputPrecision, inputScale, resultPrecision, resultScale);
+    } else {
+      return std::make_shared<DecimalRoundFunction<int128_t, int128_t>>(
+          scale, inputPrecision, inputScale, resultPrecision, resultScale);
+    }
+  }
+}
+}; // namespace
+
+std::pair<uint8_t, uint8_t>
+getResultPrecisionScale(uint8_t precision, uint8_t scale, int32_t roundScale) {
```

Comment:
maybe add a check for 'roundScale'?

Reply:
In Spark, all integers are supported to be the `roundScale` with the logic of getResultPrecisionScale.

Reply:
Updated the doc.